pkg/commands/command_helper.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020-Present VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/pivotal/build-service-cli/pkg/k8s"
+)
+
+type CommandHelper struct {
+	dryRun bool
+	output bool
+	wait   bool
+
+	outWriter io.Writer
+	errWriter io.Writer
+
+	objPrinter k8s.ObjectPrinter
+	strBuilder strings.Builder
+}
+
+func NewCommandHelper(cmd *cobra.Command) (*CommandHelper, error) {
+	dryRun, err := getBoolFlag("dry-run", cmd)
```

Comment:
This Strategy worries me slightly. What happens if the cmd doesn't have these flags set? Is there a way to guarantee that the command helper is operating on a command that is properly setup?

Reply:
the getBoolFlag() and getStringFlag() do the required checks to see if the command is configured with requested flags. mainly with this check
```
	flag := cmd.Flags().Lookup(name)
	if flag == nil {
		return false, nil
	}
```
`--wait` is configured only for the image based commands, but the `getBoolFlag("wait", cmd)` does not blow up for commands that do not have this configured Eg. clusterstack, clusterstore, builder create/patch/save

Reply:
similar conversation
https://github.com/vmware-tanzu/kpack-cli/pull/85#discussion_r496933337

Reply:
> Is there a way to guarantee that the command helper is operating on a command that is properly setup?

if required, can return error for NewCommandHelper() if none of the 3 flags are configured for a command. But I am not inclined to do it since the commandHelper is essentially harmless (as described in previous comment).